django/db/migrations/operations/models.py

```diff
@@ -716,3 +716,67 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
 
     def describe(self):
         return "Change managers on %s" % (self.name, )
+
+
+class IndexOperation(Operation):
+
+    def __init__(self, model_name, index):
+        self.model_name = model_name
+        self.index = index
+
+    def deconstruct(self):
+        kwargs = {
+            'model_name': self.model_name,
+            'index': self.index,
+        }
+        return (
+            self.__class__.__name__,
+            [],
+            kwargs,
+        )
+
+
+class AddIndex(IndexOperation):
+    """
+    Add an index on a model.
+    """
+
+    def state_forwards(self, app_label, state):
+        model_state = state.models[app_label, self.model_name.lower()]
+        if not hasattr(self.index, 'model'):
+            self.index.model = state.apps.get_model(app_label, self.model_name)
+        model_state.options['indexes'].append(self.index)
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        schema_editor.add_index(self.index)
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        schema_editor.remove_index(self.index)
+
+    def describe(self):
+        return "Create index on field(s) %s of model %s" % (
+            ', '.join(self.index.fields),
+            self.model_name,
+        )
+
+
+class RemoveIndex(IndexOperation):
```

Comment:
I think the constructor signature of this operation should only use the index name and not require an instance of an Index class